packages/auth-providers/supabase/middleware/src/index.ts

```diff
@@ -10,13 +10,15 @@ import { clearAuthState } from './util'
 
 export interface SupabaseAuthMiddlewareOptions {
   getCurrentUser: GetCurrentUser
+  extractRoles?: (decoded: any) => string[]
```

Comment:
Did you consider `getRoles` and then add that function to lib/auth like getCurrentUser is?

The verb `extract` and the use of `decoded` is odd -- I know why it is like that, but just does not feel "clean".

Reply:
Yes you're right 👍. Let me update this to `getRoles`, and what I also thought was that maybe I can implement a default one that will pull in the roles from currentUser.roles.